test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/HistoryTests.cs

```diff
@@ -236,38 +243,38 @@ public async Task GivenAQueryThatReturnsMoreThan10Results_WhenGettingSystemHisto
 
             using FhirResponse<Bundle> readResponse = await _client.SearchAsync($"_history?_tag={tag}&_since={sinceUriString}");
 
-            Assert.Equal(10, readResponse.Resource.Entry.Count);
+            AssertCount(10, readResponse.Resource.Entry);
 
             Thread.Sleep(500);
 
-            newResources.Add(await _client.CreateAsync(Samples.GetJsonSample("ObservationWithBloodPressure").ToPoco()));
-            newResources.Add(await _client.CreateAsync(Samples.GetJsonSample("ObservationWithEyeColor").ToPoco()));
+            newResources.Add(await _client.CreateByUpdateAsync(Samples.GetJsonSample("ObservationWithBloodPressure").ToPoco()));
+            newResources.Add(await _client.CreateByUpdateAsync(Samples.GetJsonSample("ObservationWithEyeColor").ToPoco()));
 
             Thread.Sleep(500);
 
             using FhirResponse<Bundle> firstBatch = await _client.SearchAsync($"_history?_tag={tag}&_since={sinceUriString}&_before={beforeUriString}");
 
-            Assert.Equal(10, firstBatch.Resource.Entry.Count);
+            AssertCount(10, firstBatch.Resource.Entry);
 
             var secondBatch = await _client.SearchAsync(firstBatch.Resource.NextLink.ToString());
 
-            Assert.Single(secondBatch.Resource.Entry);
+            AssertCount(1, secondBatch.Resource.Entry);
```

Comment:
Just for my knowldege, Does AssertCount performs better than the Assert.Single?

Reply:
No, and the implementation is below. It just includes more information when the failure occurs.